README.md

```diff
@@ -1,9 +1,137 @@
+
+```scala
+val initialContext: Context = Map(
+  factA -> 4,
+  factB -> 10
+)
+val derivations: List[Derivation] = new MyArithmics().berekeningen
+
+val resultContext: Context = FactEngine.runNormalDerivations(initialContext, derivations)
+
+println(PrettyPrinter.printContext(resultContext))
+```
+
+Executing the above code will yield the following:
+
+```
+Values in context:
+  factA = 4
+  factB = 10
+  factC = 6
+Done.
+```
+
+### Debugging
+
+If you want to see exactly what the engine is doing, you can replace the `runNormalDerivations` with `runDebugDerivations`. The return type of that function is a tuple containing the resulting `Context` and a list of `Step` objects. The latter describe exactly what actions the engine performed and why:
+
+```scala
+val initialContext: Context = Context(
+  factA -> 4,
+  factB -> 10
+)
+val derivations: List[Derivation] = new MyArithmics().berekeningen
+
+val (resultContext, steps) = FactEngine.runDebugDerivations(initialContext, derivations)
+
+println(PrettyPrinter.printSteps(steps))
+```
+
+Executing this snippet will yield an overview of the steps taken by the engine:
+
+```
+Steps taken:
+ * Evaluate: factC
+   * Result: Evaluated
+   * Change: Map(factC -> 6)
+Done.
+```
+
+For each step there is the `Fact` that was about to be evaluated (`factC`). Next is the status of the evaluation. The engine might indicate that the condition was false and thus the evaluation was skipped. Finally the `change` field shows what is added to the `Context` as a result of this evaluation.
+
+# Going Hardcore
+
+If you have complicated tasks to perform, or our DSL simply does not fit your needs, you can write custom evaluations.
+
+The DSL in the `Berekening` class yields a list of `Derivation`-objects. Constructing one yourself is not difficult, but requires some explanation:
+
+```
+case class DefaultDerivation(
+  input: List[Fact[Any]], 
+  output: Fact[Any], 
+  condition: Context => Boolean, 
+  operation: Evaluation[Any]) extends Derivation
+```
+
+For a derivation to work, the engine must know all the `Facts` it uses as input. The `input` parameter requires you to provide these `Facts` as a list.
+
+The `output` allows the engine to store the result of the derivation in the `Context`. The `output` fact is the key for this.
+
+The `condition` is a function to decide whether the `operation` should be executed. The function is provided the current `Context` containing all values known at the point of execution. Returning `true` causes the `operation` to be executed, `false` will cause it to be skipped.
+
+Finally, the `operation` is the function which will result in the value assigned to the `output` fact. You need to instantiate or extend any of the available `Evaluation` classes.
+
+# Future Work
+
+The DSL is currently only available in Dutch. We are working on translating it into English and expect this to be releasable in the next month.
```

Comment:
Here we say "in the coming month" and later we talk about the coming 2 months. It would be better to say "In Juli of 2016" for the first and "In the summer of 2016" for the second. This is a bit more vague, but atleast it does not move along with time. In December it will still be "coming Month" for the person reading it for the first time